src/Backend/OnboardingInsuranceAPI/Areas/User/UserController.cs

```diff
@@ -0,0 +1,51 @@
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using OnboardingInsuranceAPI.Extensions;
+using System.Net;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace OnboardingInsuranceAPI.Areas.User;
+
+public class UserController
+{
+    private readonly GetUser _getUser;
+    private readonly RegisterUser _registerUser;
+    private readonly UpdateUser _updateUser;
+
+    public UserController(GetUser getUser, RegisterUser registerUser , UpdateUser updateUser)
+    {
+        _getUser = getUser;
+        _updateUser = updateUser;
+        _registerUser = registerUser;
+    }
+
+    [Function("GetUser")] //read user profile by PID number
+    public  async Task<HttpResponseData> Get(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "users/{pid}")] HttpRequestData req, string pid)
+    {
```

Comment:
Thinking about this. You probably should allow anyone to just request data about everyone else. Most secure way is to only allow access to this form data to user that has the same PID as the one being requested - You can check that via JWT Token.

Reply:
This goes about all other endpoints.